### ISSUE TITLE
Ability to read Map keys

### ISSUE THREAD

**Issue by 5225225** [rust-crdt/rust-crdt#89]
I'm writing a todo list application, and the CRDT type I'm using is

`Map<uuid::Uuid, Map<String, MVReg<String, Actor>, Actor>, Actor>`

Where the outer map is a mapping from arbitrarily chosen ids to inner maps, and the inner map is a series of key=value properties about the entry.

As written, it seems like I need some external way to know which Uuids exist so I can show all the entries, since Map doesn't look like it provides access to `entries`.

I can't use `Orswot` I don't think, because I want the ability for multiple clients to modify the same item.

Would there be any issues with giving access to the keys? Could either be a Vec<K> if you have Clone for K, or an iterator over &K for any type.

**Comment by davidrusu** [comment]
Yeah we should have this, I don't see any risks if we use an interface like the following:
```rust
fn keys(&self) -> impl IntoIter<item=ReadCtx<&K,A>>;

fn values(&self) -> impl IntoIter<item=ReadCtx<&V,A>>;

fn iter(&self) -> impl IntoIter<item=ReadCtx<(&K, &V), A>>;
```

That is to say each item in the iterator should be wrapped in it's own read context.

Mind submitting a PR?

**Comment by davidrusu** [comment]
closed with #90
